FMS.Domain/Entities/Facility.cs

```diff
@@ -1,94 +1,138 @@
-        public ICollection<RetentionRecord> RetentionRecords { get; set; }
-    }
+            // The following properties only apply to Release Notifications
+            HSInumber = newFacility.HSInumber;
+            DeterminationLetterDate = newFacility.DeterminationLetterDate;
+            Comments = newFacility.Comments;
+            PreRQSMcleanup = newFacility.PreRQSMcleanup;
+            ImageChecked = newFacility.ImageChecked;
+            DeferredOnSiteScoring = newFacility.DeferredOnSiteScoring;
+            AdditionalDataRequested = newFacility.AdditionalDataRequested;
+            VRPReferral = newFacility.VRPReferral;
+            HasERecord = newFacility.HasERecord;
+            RNDateReceived = newFacility.RNDateReceived;
+            HistoricalUnit = newFacility.HistoricalUnit;
+            HistoricalComplianceOfficer = newFacility.HistoricalComplianceOfficer;
+            TaxId = newFacility.TaxId;
+        }
+
+        // Existing ID for Facility May be used by Programs - System Generated, but not a Guid
+        public string FacilityNumber { get; set; }
+
+        // File label and Cabinet where this Facility is located
+        public Guid FileId { get; set; }
+        public File File { get; set; }
+
+        // Type of Facility
+        public Guid? FacilityTypeId { get; set; }
+        public FacilityType FacilityType { get; set; }
+
+        // Unit overseeing this Facility
+        public Guid? OrganizationalUnitId { get; set; }
+        public OrganizationalUnit OrganizationalUnit { get; set; }
+
+        // Program Budget for this Facility
+        public Guid? BudgetCodeId { get; set; }
+        public BudgetCode BudgetCode { get; set; }
+
+        // Facility Name
+        public string Name { get; set; }
+
+        // Compliance Officer assigned to this Facility
+        public Guid? ComplianceOfficerId { get; set; }
+        public ComplianceOfficer ComplianceOfficer { get; set; }
+
+        public Guid? FacilityStatusId { get; set; }
+        public FacilityStatus FacilityStatus { get; set; }
+
+        // Location description distinct from mailing address
+        public string Location { get; set; }
+
+        // Facility Address
+        [StringLength(100)]
+        public string Address { get; set; }
+
+        [StringLength(30)]
+        public string City { get; set; }
+
+        [StringLength(20)]
+        public string State { get; set; }
+
+        [StringLength(10)]
+        public string PostalCode { get; set; }
+
+        // site Coordinates
+        [Column(TypeName = "decimal(8, 6)")]
+        public decimal Latitude { get; set; }
+
+        [Column(TypeName = "decimal(9, 6)")]
+        public decimal Longitude { get; set; }
+
+        public int CountyId { get; set; }
+
+        public County County { get; set; }
+
+        public bool IsRetained { get; set; } = true;
+
+        // The following properties only apply to Release Notifications

```

Comment:
Excellent point that I have considered. My question to you is, since these fields only apply to release notifications, does it matter if they are marked as "false" instead of "null"?? It makes no difference from a programmatic standpoint. Or does it?

Reply:
Doug, my fear is that they will edit an existing Non-Release Notification, set the dropdown for ""Type" to "RN Release Notification", the aforementioned checkboxes will appear, and the HTML/ASP will not know how to handle a null value for a checkbox and crash the page accidentally. Is this a potential problem, or am I mistaken?

Reply:
> Excellent point that I have considered. My question to you is, since these fields only apply to release notifications, does it matter if they are marked as "false" instead of "null"?? It makes no difference from a programmatic standpoint. Or does it?

It might not make a difference in the end, but to me storing a value - even a default value - when no data was entered by the user creates ambiguity. A better question would be is there a compelling reason to not make the fields nullable? 

> Doug, my fear is that they will edit an existing Non-Release Notification, set the dropdown for ""Type" to "RN Release Notification", the aforementioned checkboxes will appear, and the HTML/ASP will not know how to handle a null value for a checkbox and crash the page accidentally. Is this a potential problem, or am I mistaken?

ASP.NET should be able to handle null values, but the web pages use DTOs, not entities. The properties of the DTO don't necessarily have to be nullable depending on your needs. But when mapping the DTO back to the entity to store in the database, you can use the facility type to determine whether nulls or other default values get stored.